<?xml version="1.0"?>
<launch>
  <arg name="wrench_topic_name" default="/external_wrench"/>
  <arg name="command_topic_name" default="/planner_command"/>
  <arg name="state_topic_name" default="/ur_robot_state"/>
  <arg name="robot_urdf_name" default="$(find ur_controllers)/config/ur5.urdf"/>

  <arg name="robot_ip" default="192.168.1.100"/>
  <include file="$(find ur_robot_driver)/launch/ur5_bringup.launch">
    <arg name="robot_ip" value="$(arg robot_ip)"/>
  </include>
  
  <rosparam command="load" file="$(find ur_controllers)/config/ur5_admittance_params.yaml"/>
  <node name="joint_admittance_controller" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="joint_admittance_controller">
    <param name="wrench_topic_name" value="$(arg wrench_topic_name)"/>
    <param name="command_topic_name" value="$(arg command_topic_name)"/>
    <param name="state_topic_name" value="$(arg state_topic_name)"/>
    <param name="robot_urdf_name" value="$(arg robot_urdf_name)"/>
  </node>
</launch>